Tasks/AzureFunctionAppV1/deploymentProvider/ConsumptionWebAppDeploymentProvider.ts

```diff
@@ -0,0 +1,37 @@
+import { AzureRmWebAppDeploymentProvider } from './AzureRmWebAppDeploymentProvider';
+import tl = require('vsts-task-lib/task');
+import { AzureAppService } from 'azurermdeploycommon/azure-arm-rest/azure-arm-app-service';
+import { AzureAppServiceUtility } from 'azurermdeploycommon/operations/AzureAppServiceUtility';
+
+export class ConsumptionWebAppDeploymentProvider extends AzureRmWebAppDeploymentProvider {
+
+    public async PreDeploymentStep() {
+        this.appService = new AzureAppService(this.taskParams.azureEndpoint, this.taskParams.ResourceGroupName, this.taskParams.WebAppName, 
+            this.taskParams.SlotName, this.taskParams.WebAppKind);
+        this.appServiceUtility = new AzureAppServiceUtility(this.appService);
+    }
+ 
+    public async DeployWebAppStep() {
+        let appSettings = await this.appService.getApplicationSettings();
```

Comment:
Can we have this logic in separate function?

Reply:
It is done.